Default acceptance test in Spock makes the pipeline runs forever

**Describe the bug**
The current acceptance test for the Spock Quickstarter is making the pipeline run continuously.

**To Reproduce**
Steps to reproduce the behavior:
1. Execute Spock default acceptance test
4. See problem

**Expected behavior**
The pipeline should finish even if the tests fails

**Screenshots**
![image](https://user-images.githubusercontent.com/26645694/153390276-9fcb0eba-58b6-4b0c-a403-3ce5de3fb18a.png)


**Affected version (please complete the following information):**
 - OpenShift: 3.11
 - OpenDevStack 4.x

**Log Output with debug **
```
10:00:26.509 [DEBUG] [TestEventLogger] Gradle Test Run :acceptance-groovy STARTED
10:00:26.522 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector] test-class-scan : failed to scan parent class java/lang/Object, could not find the class file
10:00:26.523 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector] test-class-scan : failed to scan parent class java/lang/Object, could not find the class file
10:00:26.529 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager] extracted class groovy/lang/Closure from groovy-2.5.6.jar
10:00:26.537 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager] extracted class groovy/lang/GroovyObjectSupport from groovy-2.5.6.jar
10:00:26.537 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector] test-class-scan : failed to scan parent class java/lang/Object, could not find the class file
10:00:26.554 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager] extracted class geb/Page from geb-core-3.2.jar
10:00:26.555 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.AbstractTestFrameworkDetector] test-class-scan : failed to scan parent class java/lang/Object, could not find the class file
10:00:26.556 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager] extracted class geb/spock/GebReportingSpec from geb-spock-3.2.jar
10:00:26.557 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager] extracted class geb/spock/GebSpec from geb-spock-3.2.jar
10:00:26.587 [DEBUG] [org.gradle.api.internal.tasks.testing.detection.ClassFileExtractionManager] extracted class spock/lang/Specification from spock-core-1.3-groovy-2.5.jar
10:00:26.588 [DEBUG] [org.gradle.internal.work.DefaultWorkerLeaseService] Worker lease root.1.6.219 started (1 worker(s) in use).
10:00:26.588 [DEBUG] [org.gradle.internal.resources.AbstractTrackedResourceLock] Dispatch org.gradle.api.internal.tasks.testing.processors.RestartEveryNTestClassProcessor@3ad60ab2: acquired lock on root.1.6.219
10:00:26.592 [DEBUG] [org.gradle.internal.remote.internal.inet.TcpIncomingConnector] Listening on [8986d166-4ecd-4218-8d18-3e68d95667e2 port:45577, addresses:[/0:0:0:0:0:0:0:1%lo, /127.0.0.1]].
10:00:26.592 [DEBUG] [org.gradle.process.internal.worker.DefaultWorkerProcessBuilder] Creating Gradle Test Executor 7
10:00:26.592 [DEBUG] [org.gradle.process.internal.worker.child.WorkerProcessClassPathProvider] Using worker process classpath: [/home/jenkins/.gradle/caches/5.6.4/workerMain/gradle-worker.jar]
10:00:26.594 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: STARTING
10:00:26.594 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Waiting until process started: Gradle Test Executor 7.
10:00:26.596 [DEBUG] [org.gradle.process.internal.DefaultExecHandle] Changing state to: STARTED
10:00:26.596 [DEBUG] [org.gradle.process.internal.ExecHandleRunner] waiting until streams are handled...
10:00:26.596 [INFO] [org.gradle.process.internal.DefaultExecHandle] Successfully started process 'Gradle Test Executor 7'
10:00:26.602 [ERROR] [system.err] Picked up JAVA_TOOL_OPTIONS: -XX:+UnlockExperimentalVMOptions -Dsun.zip.disableMemoryMapping=true
10:00:27.040 [DEBUG] [TestEventLogger] 
10:00:27.040 [DEBUG] [TestEventLogger] Gradle Test Executor 7 STARTED
10:00:27.048 [DEBUG] [TestEventLogger] 
10:00:27.048 [DEBUG] [TestEventLogger] DemoAcceptance STARTED
10:00:27.137 [ERROR] [system.err] WARNING: An illegal reflective access operation has occurred
10:00:27.137 [ERROR] [system.err] WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v7.Java7$1 (file:/home/jenkins/.gradle/caches/modules-2/files-2.1/org.codehaus.groovy/groovy/2.5.6/6936e700f0fb1b50bac0698ada4347a769d40199/groovy-2.5.6.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
10:00:27.137 [ERROR] [system.err] WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v7.Java7$1
10:00:27.137 [ERROR] [system.err] WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
10:00:27.137 [ERROR] [system.err] WARNING: All illegal access operations will be denied in a future release
10:00:28.580 [DEBUG] [TestEventLogger] 
10:00:28.580 [DEBUG] [TestEventLogger] DemoAcceptance > goes to W3C editor STARTED
10:00:31.036 [DEBUG] [TestEventLogger] 
10:00:31.036 [DEBUG] [TestEventLogger] DemoAcceptance > goes to W3C editor STANDARD_ERROR
10:00:31.037 [DEBUG] [TestEventLogger]     Feb 10, 2022 10:00:31 AM com.gargoylesoftware.htmlunit.DefaultCssErrorHandler error
10:00:31.037 [DEBUG] [TestEventLogger]     WARNING: CSS error: 'https://www.w3schools.com/lib/codemirror.css' [26:1] Error in style rule. (Invalid token ".". Was expecting one of: <S>, <LBRACE>, <COMMA>.)
10:00:31.037 [DEBUG] [TestEventLogger]     Feb 10, 2022 10:00:31 AM com.gargoylesoftware.htmlunit.DefaultCssErrorHandler warning
10:00:31.037 [DEBUG] [TestEventLogger]     WARNING: CSS warning: 'https://www.w3schools.com/lib/codemirror.css' [26:1] Ignoring the following declarations in this rule.
10:00:31.181 [DEBUG] [TestEventLogger]     Feb 10, 2022 10:00:31 AM com.gargoylesoftware.htmlunit.IncorrectnessListenerImpl notify
10:00:31.182 [DEBUG] [TestEventLogger]     WARNING: Obsolete content type encountered: 'text/javascript'.
10:00:31.421 [LIFECYCLE] [org.gradle.process.internal.health.memory.MemoryManager] 
10:00:31.421 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting OS memory status event {Total: 67368652800, Free: 55317757952}
10:00:31.421 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting JVM memory status event {Maximum: 1073741824, Committed: 155189248}
10:00:32.187 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting JVM memory status event {Maximum: 536870912, Committed: 160432128}
10:00:36.421 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting OS memory status event {Total: 67368652800, Free: 55301922816}
10:00:36.421 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting JVM memory status event {Maximum: 1073741824, Committed: 155189248}
10:00:37.187 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting JVM memory status event {Maximum: 536870912, Committed: 160432128}
10:00:41.421 [DEBUG] [org.gradle.process.internal.health.memory.MemoryManager] Emitting OS memory status event {Total: 67368652800, Free: 55328407552}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@
 - Scala agent updated from Jenkins base image changes ([#721](https://github.com/opendevstack/ods-quickstarters/issues/721))
 - terraform agent updated from Jenkins base image changes ([#724](https://github.com/opendevstack/ods-quickstarters/issues/724))
 - Bump antora page version in master (https://github.com/opendevstack/ods-documentation/issues/66)
+- Default acceptance test in Spock makes the pipeline runs forever ([#706](https://github.com/opendevstack/ods-quickstarters/issues/706))
 
 ### Modified
 

--- a/e2e-spock-geb/files/build.gradle
+++ b/e2e-spock-geb/files/build.gradle
@@ -58,24 +58,24 @@ repositories {
 }
 
 ext {
-    junitVersion = "5.5.2"
-    spockVersion = "1.3-groovy-2.5"
-    gebVersion = "3.2"
-    seleniumVersion = "3.141.59"
-    htmlunitVersion = "2.36.0"
-    unirestVersion = "3.2.00"
+    junitVersion = "5.8.2"
+    spockVersion = "2.1-groovy-3.0"
+    gebVersion = "5.1"
+    seleniumVersion = "4.1.2"
+    htmlunitVersion = "3.58.0"
+    unirestVersion = "3.13.6"
     // When a test fail we can continue or fail the stage
     CONTINUE_WHEN_TEST_FAIL = true
 }
 
 dependencies {
-    testCompile "org.junit.jupiter:junit-jupiter-engine:${junitVersion}"
-    testCompile "org.spockframework:spock-core:${spockVersion}"
-    testCompile "com.konghq:unirest-java:${unirestVersion}"
-    testCompile "org.gebish:geb-spock:${gebVersion}"
-    testCompile "org.seleniumhq.selenium:selenium-firefox-driver:${seleniumVersion}"
-    testCompile "org.seleniumhq.selenium:htmlunit-driver:${htmlunitVersion}"
-    testRuntime "org.seleniumhq.selenium:selenium-support:${seleniumVersion}"
+    testImplementation "org.junit.jupiter:junit-jupiter-engine:${junitVersion}"
+    testImplementation "org.spockframework:spock-core:${spockVersion}"
+    testImplementation "com.konghq:unirest-java:${unirestVersion}"
+    testImplementation "org.gebish:geb-spock:${gebVersion}"
+    testImplementation "org.seleniumhq.selenium:selenium-firefox-driver:${seleniumVersion}"
+    testImplementation "org.seleniumhq.selenium:htmlunit-driver:${htmlunitVersion}"
+    testImplementation "org.seleniumhq.selenium:selenium-support:${seleniumVersion}"
 }
 
 sourceSets {
@@ -89,8 +89,8 @@ sourceSets {
         resources {
             srcDir 'src/test/resources'
         }
-        compileClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntime
-        runtimeClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntime
+        compileClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntimeClasspath
+        runtimeClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntimeClasspath
     }
 
     integration {
@@ -103,8 +103,8 @@ sourceSets {
         resources {
             srcDir 'src/test/resources'
         }
-        compileClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntime
-        runtimeClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntime
+        compileClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntimeClasspath
+        runtimeClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntimeClasspath
     }
 
     acceptance {
@@ -117,20 +117,20 @@ sourceSets {
         resources {
             srcDir 'src/test/resources'
         }
-        compileClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntime
-        runtimeClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntime
+        compileClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntimeClasspath
+        runtimeClasspath += sourceSets.main.output + sourceSets.test.output + configurations.testRuntimeClasspath
     }
 }
 
 configurations {
-    installationCompile.extendsFrom testCompile
-    installationRuntime.extendsFrom testRuntime
+    installationCompile.extendsFrom testImplementation
+    installationRuntime.extendsFrom testRuntimeClasspath
 
-    integrationCompile.extendsFrom testCompile
-    integrationRuntime.extendsFrom testRuntime
+    integrationCompile.extendsFrom testImplementation
+    integrationRuntime.extendsFrom testRuntimeClasspath
 
-    acceptanceCompile.extendsFrom testCompile
-    acceptanceRuntime.extendsFrom testRuntime
+    acceptanceCompile.extendsFrom testImplementation
+    acceptanceRuntime.extendsFrom testRuntimeClasspath
 }
 
 class TestExecutionPhases {
@@ -154,10 +154,7 @@ def executeTest(def type, def language) {
         description = "Runs ${type} tests ${language}."
         group = "verification"
 
-        if ("${language}".equalsIgnoreCase(TestLanguages.JAVA)) {
-            // Mandatory to run JUnit 5 test but incompatible with groovy tests
-            useJUnitPlatform()
-        }
+        useJUnitPlatform()
 
         ignoreFailures = "${CONTINUE_WHEN_TEST_FAIL}"
         testClassesDirs = sourceSets["${type}"].output.classesDirs

--- a/e2e-spock-geb/files/gradle/wrapper/gradle-wrapper.properties
+++ b/e2e-spock-geb/files/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/e2e-spock-geb/files/src/test/acceptance/groovy/DemoAcceptanceSpec.groovy
+++ b/e2e-spock-geb/files/src/test/acceptance/groovy/DemoAcceptanceSpec.groovy
@@ -2,20 +2,20 @@ import geb.Page
 import geb.spock.GebReportingSpec
 import spock.lang.Stepwise
 
-class W3CAcceptanceHomePage extends Page {
-    static url = "/html/tryit.asp?filename=tryhtml_basic_paragraphs"
-    static at = { title == "Tryit Editor v3.7" }
+class GitHubAcceptanceHomePage extends Page {
+    static url = "/opendevstack/ods-quickstarters"
+    static at = { title.contains("quickstarters")}
 }
 
 @Stepwise
 class DemoAcceptance extends GebReportingSpec {
 
-    def "goes to W3C editor"() {
-        given: "User goes to we3 editor for paragraphs and checks the content"
-        to W3CAcceptanceHomePage
-        sleep (2000)
+    def "goes to GH ods-quickstarters"() {
+        given: "User goes to ods-quickstarters and checks the content"
+        to GitHubAcceptanceHomePage
+
         // print evidence of two fields (the input area and iframe content)
-        SpecHelper.printEvidenceForPageElement(this, 1, $("#textareaCode"), "code area")
+        SpecHelper.printEvidenceForPageElement(this, 1, $("[data-content='Code']"), "code area")
         SpecHelper.printEvidenceForPageElement(this, 1, $("#iframecontainer"), "rendered code area")
 
         // print the two evidence fields through map

--- a/e2e-spock-geb/files/src/test/resources/GebConfig.groovy
+++ b/e2e-spock-geb/files/src/test/resources/GebConfig.groovy
@@ -8,7 +8,7 @@ def properties = new SpecHelper().getApplicationProperties()
 
 // Selenium driver (True in constructor to use JavaScript)
 driver = {
-    HtmlUnitDriver driver = new HtmlUnitDriver(BrowserVersion.CHROME, true) {
+    HtmlUnitDriver driver = new HtmlUnitDriver(BrowserVersion.BEST_SUPPORTED, true) {
       @Override
       protected WebClient newWebClient(BrowserVersion version) {
           WebClient webClient = super.newWebClient(version);

--- a/e2e-spock-geb/files/src/test/resources/application.properties
+++ b/e2e-spock-geb/files/src/test/resources/application.properties
@@ -1,5 +1,5 @@
 # Base url with the application to test - w3c schools
-config.application.url=https://www.w3schools.com
+config.application.url=https://github.com
 
 # Base url with the application to test - OpenShift app using env vars from Jenkinsfile
 # config.application.url=http://app-${OPENSHIFT_PROJECT}.${OPENSHIFT_APP_DOMAIN}

